### PR TITLE
Merged view with live switch and group checking

### PR DESCRIPTION
**Pull request by Polej**

### CODE REVIEW

File: src/components/Pages/Admin/View.vue [hunk 1]
```diff
@@ -216,16 +234,57 @@ export default {
...
+        },
     },
+
     async created() {
         await this.findClasses();
     },
 
+    watch: {
+        liveMergeSwitch(newVal) {
+            if (newVal) this.GeogebraViews.mergeGoLive();
```
Comment (IwoHerka) [inline src/components/Pages/Admin/View.vue]: @Polej Rewrite this to use proper block (curly braces).

File: src/components/Pages/Admin/View.vue [hunk 2]
```diff
@@ -216,16 +234,57 @@ export default {
...
+                this.$log.debug('this.showMenuBar', this.showMenuBar);
+                this.$log.debug('this.$refs.ibox_content.clientWidth',
+                    this.$refs.ibox_content.clientWidth);
+
+                this.GeogebraViews.mergeViews(
+                    this.selectedGroups.map(g => g._id),
+                    { // params for geogebra
+                        showMenubar: this.showMenuBar,
+                        width: (this.$refs.ibox_content.clientWidth - 60),
+                        height: (this.$refs.ibox_content.clientWidth - 60) * 2 / 3,
```
Comment (IwoHerka) [inline src/components/Pages/Admin/View.vue]: This is magic as hell. Try to provide some explanation why you do those calculations in the comment.

Also: refactor out `this.$refs.ibox_content.clientWidth - 60`

File: src/components/Pages/Admin/View.vue [hunk 3]
```diff
@@ -216,16 +234,57 @@ export default {
 
                 this.GeogebraViews = new GeogebraViews(this.groupsInClass, {
                     log: this.$log,
-                    width: this.$refs.ibox_content.clientWidth - 60,
+                    width: 400, // this.$refs.ibox_content.clientWidth - 60,
```
Comment (IwoHerka) [inline src/components/Pages/Admin/View.vue]: Why do you hardcode those values? Isnt 400x400 small?

File: src/components/Pages/Admin/View.vue [hunk 4]
```diff
@@ -185,19 +192,30 @@ export default {
...
 
             if (!this.groupsInClass.length) {
                 this.showToast('Selected class nas no groups', 'warning');
             } else {
-                this.groupsInClass.forEach((g, i) => {
-                    this.groupsInClass[i].domId = `ggb_applet_${g._id}`;
+                this.classSelected = true;
+
+                this.groupsInClass.forEach((group, idx) => {
+                    this.groupsInClass[idx].domId = `ggb_applet_${group._id}`;
```
Comment (IwoHerka) [inline src/components/Pages/Admin/View.vue]: Dont hardcode IDs! Move this to util method `getAppletId` or something similar.

File: src/components/Geogebra/GeogebraViews.js [hunk 5]
```diff
@@ -74,15 +75,43 @@ export default class {
...
+            log: this.log,
+        }, workshopIds);
+
+        this.mergedView.inject();
+    }
+
+    mergeGoLive() {
+        this.log.debug('Going live');
+        this.mergedView.clear();
+        this.mergedView.loadWorkshops();
```
Comment (IwoHerka) [inline src/components/Geogebra/GeogebraViews.js]: Shouldnt `initializeCallbacks` be before loading workhops? What will happen in an unlikely scenario that workshops will arrive before callbacks are set?

File: src/components/Geogebra/GeogebraViews.js [hunk 6]
```diff
@@ -74,15 +75,43 @@ export default class {
...
         });
+    }
 
-        api.service('workshops').on('created', (workshop) => {
-            console.log('Workshop created', workshop);
-        });
+    mergeViews(workshopIds, params) {
+        this.mergedView = new GeogebraMergedAdminView({
+            container: 'merged_ggb_applet',
+            id: 'merged_ggb_applet',
```
Comment (IwoHerka) [inline src/components/Geogebra/GeogebraViews.js]: Dont hardcode IDs!

File: src/components/Geogebra/GeogebraMergedAdminView.js [hunk 7]
```diff
@@ -0,0 +1,452 @@
+import feathersClient from '../../feathers-client';
```
Comment (IwoHerka) [inline src/components/Geogebra/GeogebraMergedAdminView.js:1]: This code looks copied over, where is it from?

**Review by IwoHerka — CHANGES_REQUESTED** [review]

File: src/components/Pages/Admin/View.vue (hunk 3 above)
Comment (Polej) [reply src/components/Pages/Admin/View.vue]: Client requested smaller tiles
https://makimo.atlassian.net/browse/MATHNETNG-105

File: src/components/Geogebra/GeogebraMergedAdminView.js (hunk 7 above)
Comment (Polej) [reply src/components/Geogebra/GeogebraMergedAdminView.js:1]: It is taken from one of first versions of student's client or listener

File: src/components/Geogebra/GeogebraViews.js (hunk 5 above)
Comment (Polej) [reply src/components/Geogebra/GeogebraViews.js]: You may be right, it may probably omit some events in-between

**Review by IwoHerka — APPROVED** [review]

**Pull request merged**